### DISCUSSION TITLE
On performance optimization

### DISCUSSION THREAD

**Discussion by thienvc** [dnnsoftware/Dnn.Platform#6239] [General]
After logging in, the following code slowed down the web performance
https://github.com/dnnsoftware/Dnn.Platform/blob/2e7a5eb203941b590a5dba9e820bd286e5b2a9e8/DNN%20Platform/Website/admin/Menus/ModuleActions/ModuleActions.ascx.cs#L116
We should save the "QS_FirstLoad" setting somewhere other than in "ModuleSettings"
Because this action leads to calling ClearCache(tabid), making the cache useless

**Comment by mitchelsellers** [comment]
Looking at the code, I think that this is something that SHOULD be fixed, as this for sure is a performance "oops" In my opinion.

I'm going to submit a PR suggestion

**Reply by mitchelsellers** [reply]
@thienvc This is a great find, I did confirm the performance issue and submitted a fix here - https://github.com/dnnsoftware/Dnn.Platform/pull/6243
